v2/concordium/types.proto

```diff
@@ -2168,6 +2168,205 @@ message ElectionInfo {
+          // The peer does not have any data unknown to us. If we receive a message from the
+          // peer that refers to unknown data (e.g., an unknown block) the peer is marked as pending.
+          UPTODATE = 0;
+          // The peer might have some data unknown to us. A peer can be in this state either because
+          // it sent a message that refers to data unknown to us, or before we have established a baseline with it.
+          // The latter happens during node startup, as well as upon protocol updates until the initial catchup handshake
+          // completes.
+          PENDING = 1;
+          // The node is currently catching up by requesting blocks from this peer.
+          // There will be at most one peer with this status at a time.
+          // Once the peer has responded to the request, its status will be changed to:
+          // - 'UPTODATE' if the peer has no more data that is not known to us
+          // - 'PENDING' if the node has more data that is unknown to us.
+          CATCHINGUP = 2;
+      }
+      // Network statistics for the peer
+      message NetworkStats {                
+        // The number of messages sent to the peer.
+        uint64 packets_sent = 2;
+    
+        // The number of messages received from the peer.
+        uint64 packets_received = 3;
+    
+        // The connection latency (i.e., ping time) in milliseconds.
+        uint64 latency = 4;
+      }
+
+      // The identifier of the peer that it
+      // wishes to be identified by.
+      PeerId peer_id = 1;
+      // The port of the peer.
+      IpSocketAddress socket_address = 2;
+      // Network related statistics for the peer.
+      NetworkStats network_stats = 3;
+      // consensus related information of the peer.
+      oneof consensus_info {
+        // The peer is of type `Bootstrapper` is not participating in consensus 
+        // and thus has no catchup status.
+        Empty bootstrapper = 4;
+        // The peer is a regular node and have 
+        // an associated catchup status.
+        CatchupStatus node_catchup_status = 5;
+      }
+  }
+
+  repeated Peer peers = 1;
+}
+
+// Node info response
+// Contains various information of the
+// enquired node.
+message NodeInfo {
+  // Network related information of the node.
+  message NetworkInfo {   
+    // The node id.
+    PeerId node_id = 1;
+    // Total number of packets sent by the node.
+    uint64 peer_total_sent = 2;
+    // Total number of packets received by the node.
+    uint64 peer_total_received = 3;
+    // Average outbound throughput in bytes per second.
+    uint64 avg_bps_in = 4;
+    // Average inbound throughput in bytes per second.
+    uint64 avg_bps_out = 5;
+  }
+
+  // Consensus info for a node configured with baker keys.
+  message BakerConsensusInfo {
+    // The committee information of a node configured with 
+    // baker keys but somehow the node is part of the 
+    // baking committee.
+    enum PassiveCommitteeInfo {
```

Comment:
This documentation seems strange. Not sure what "but somehow" ... part means.

Reply:
Not'ed it.